autogpt_platform/backend/backend/executor/manager.py

```diff
@@ -256,10 +263,12 @@ def update_execution(status: ExecutionStatus) -> ExecutionResult:
 
         # Update execution stats
         if execution_stats is not None:
-            execution_stats.update(node_block.execution_stats)
-            execution_stats["input_size"] = input_size
-            execution_stats["output_size"] = output_size
-            execution_stats["cost"] = cost
+            execution_stats = execution_stats.model_copy(
```

Comment:
why do we need to copy?

Reply:
I'm copying over the existing behavior of updating the previous with the new data. This is how you do it in pydantic as far as I know